docs/kb/semgrep-code/finding_all_taints.md

```diff
@@ -24,44 +24,29 @@ File 1 function4(sourceA)
+### Interfile analysis
 
-No configuration can expose the the 4 --> 1 traversal. Instead, it's a case of achieving cardinality here, placing the same type of taint in the same bucket regardless of path traversal. This affords much more efficient and intelligible findings triage: you are inspecting only unique findings. This is especially useful for languages with polymorphic classes that can add noise for a singleton taint.  
+If both data flows are identified in the same scan, and the scan has interfile analysis enabled (`--pro`, or Pro Engine enabled in the Cloud Platform) only Call stack 1 is reported as a finding. It has a shorter path, and has the same sourceA -> sinkB taint.  
 
-### Some assumptions
+This speeds up triage by ensuring you are only reviewing unique findings. It's especially useful for languages with polymorphic classes that can add noise for a singleton taint.  
 
-* Interfile analysis is used: the `--pro` flag is specified for either the`semgrep scan` or `semgprep ci` command.
-* If only intrafile or interprocedural analysis is performed (`--pro-intrafile`), Semgrep would have reported the taint against call stack 2.
+### Intrafile analysis
 
-Be mindful of the scope you are specifying for the scan, from the `semgrep scan --help` or `semgrep ci --help` pages:
-
-```
-Semgrep Pro Engine options: 
-    --pro                         Inter-file analysis and Pro languages (currently just
-                                  Apex). Requires Semgrep Pro Engine, contact
-                                  support@semgrep.com for more information on this.
-    --pro-intrafile               Intra-file inter-procedural taint analysis. Implies
-                                  --pro-languages. Requires Semgrep Pro Engine, contact
-                                  support@semgrep.com. for more information on this.
-```
+If only intrafile / interprocedural analysis is performed (`--pro-intrafile`), Semgrep only reports a finding for call stack 2. Call stack 1 would not be identified, because it crosses file boundaries.
 
 ## Best practices for testing tainted data flows
 
-In the early phases of your Semgrep rollout, the Semgrep team recommends to test against all your test cases for interfile coverage in taint mode, that the Semgrep engine can indeed track a source to a sink transcending across files and functions.  To completely exhaust the scope, start with the shortest path and then change the implementation such that the source is no longer in the scope of the shortest path and repeat this until you have achieved the longest, targeted path traversal. This is assuming you are also specifying to `--pro` to cover the greatest atomicity across files.  
+If you want to understand in greater detail how Semgrep is detecting tainted data flows, you can use existing or constructed test cases to review the different paths. 
```

Comment:
```suggestion
To understand in greater detail how Semgrep detects tainted data flows, you can use your own test cases to review different paths.
```
Removed "existing or constructed" as it adds to the confusion. Existing? Where? From me or Semgrep, Inc?
Also, technically, all test cases are "constructed" so I think we can do without it also. (Just a suggestion).

Reply:
yeah I agree, I like this revision. It's not letting me apply it because of being outdated, so I'll do it from the CLI and then merge!